### ISSUE TITLE
New Rule: Enforce return in getter

### ISSUE THREAD

**Issue by jaredmcateer** [eslint/eslint#8449]
**Please describe what the rule should do:**

Enforces that a return statement is present in property getters

**What category of rule is this? (place an "X" next to just one item)**

[ ] Enforces code style
[X] Warns about a potential error
[ ] Suggests an alternate way of doing something
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**

```js

/*eslint getter-return: "error"*/

// Incorrect
class MyClass {
  get a() {
     this.retreiveVal('a');
  }
}

function MyConstructorFunction() {
  Object.defineProperty(this, 'a', {
    get: function () {
      this.retreiveVal('a');
    }
  });
}


// Correct
class MyClass {
  get a() {
     return this.retreiveVal('a');
  }
}

function MyConstructorFunction() {
  Object.defineProperty(this, 'a', {
    get: function () {
       return this.retreiveVal('a');
    }
  });
}
```

**Why should this rule be included in ESLint (instead of a plugin)?**

I cannot think of a valid use case why someone would want to define a getter that did not return a value and it can be easily overlooked when reviewing a long list of getter/setters. I think the problem is generic enough that it should be included by ESLint.

**Comment by platinumazure** [comment]
I'm :+1: to this idea. I think we should for sure do this in getter methods. I'm less sure about Object.defineProperty and friends just due to static analysis limitations, but I'm not opposed to adding it there either.

**Comment by platinumazure** [comment]
@jaredmcateer Would you be interested in writing a PR if this gets accepted?

**Comment by jaredmcateer** [comment]
I am travelling for a few weeks but I would definitely be up for writing
this rule when I am back and have the time.

**Comment by ilyavolodin** [comment]
Looks like we have enough votes to accept this. Just need a champion. Anyone willing to champion this issue?

**Comment by platinumazure** [comment]
I'll champion.

@jaredmcateer Let us know when you get a chance to start work on this, or if your schedule fills up and it looks like you won't be able to do it. No rush, and enjoy your travels!

**Comment by Aladdin-ADD** [comment]
working on this. maybe the following weekend.
